script/check-patch-diff.js

```diff
@@ -0,0 +1,16 @@
+const { spawnSync } = require('child_process');
+const path = require('path');
+const { inspect } = require('util');
+
+const srcPath = path.resolve(__dirname, '..', '..', '..');
+const patchExportPath = path.resolve(__dirname, 'export_all_patches.py');
```

Comment:
One potential gotcha I can imagine happening here is if someone manually adds a patch to `patches/` without going through the `export-patches` flow—this would delete that patch. Then the `git add` would commit the deletion to the index.

It seems a bit rude to delete the file that someone just added. Perhaps we could skip the `git add` and just fail the lint step if there are changes as a result of `export_all_patches`?

Reply:
oh - good point 🤔 i agree - we maybe shouldn't be nuking things out from underneath users with no warning, but on the other hand i guess it does force use of what arguably should be done anyway? Lint-staged typically doesn't fail that way, but i'm definitely open to it

Reply:
>  and just fail the lint step if there are changes as a result of export_all_patches?

Yeah I think we should do this, let users figure out what to do with their _bad things_, would suck to lose a change because you didn't sync or run export or something